lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1357,6 +1398,278 @@ class DifferentiableActivityInfo {
+        // Check for active 'inout' arguments.
+        bool isInout = false;
+        auto paramInfos = ai->getSubstCalleeConv().getParameters();
+        for (unsigned i : swift::indices(paramInfos)) {
+          if (paramInfos[i].isIndirectInOut() &&
+              activityInfo.isActive(ai->getArgumentsWithoutIndirectResults()[i],
+                                    indices)) {
+            // Reject functions with active inout arguments. It's not yet supported.
+            isInout = true;
+            break;
+          }
+        }
+        if (isInout)
+          break;
+
+         // Add linear map to struct for active instructions.
+        // Do not add it for array functions since those are already linear
+        // and we don't need to add it to the struct.
+        if (shouldBeDifferentiated(ai, indices) &&
+            !ai->hasSemantics("array.uninitialized_intrinsic")) {
+          SmallVector<SILValue, 4> allResults;
+          allResults.push_back(ai);
+          allResults.append(ai->getIndirectSILResults().begin(),
+                            ai->getIndirectSILResults().end());
+
+           // Check if there are any active results or arguments. If not, skip
+          // this instruction.
+          auto hasActiveResults = llvm::any_of(
+              allResults, [&](SILValue res) {
+            return activityInfo.isActive(res, indices);
+          });
+          auto hasActiveArguments = llvm::any_of(
+              ai->getArgumentsWithoutIndirectResults(), [&](SILValue arg) {
+            return activityInfo.isActive(arg, indices);
+          });
+          if (!hasActiveResults || !hasActiveArguments)
+            continue;
+
+           unsigned source;
+          AutoDiffIndexSubset *parameters;
+
+           SmallVector<unsigned, 8> activeParamIndices;
+          SmallVector<unsigned, 8> activeResultIndices;
+          collectMinimalIndicesForFunctionCall(
+              ai, allResults, indices, activityInfo, activeParamIndices,
+              activeResultIndices);
+          source = activeResultIndices.front();
+
+           // If function is already marked differentiable, differentiate WRT
+          // all parameters.
+          auto originalFnSubstTy = ai->getSubstCalleeType();;
+          if (originalFnSubstTy->isDifferentiable()) {
+            parameters = originalFnSubstTy->getDifferentiationParameterIndices();
+          } else {
+            parameters = AutoDiffIndexSubset::get(
+                original->getASTContext(),
+                ai->getArgumentsWithoutIndirectResults().size(),
+                activeParamIndices);
+          }
+          SILAutoDiffIndices curIndices(activeResultIndices.front(),
+          AutoDiffIndexSubset::get(
+              builder.getASTContext(),
+              ai->getArgumentsWithoutIndirectResults().size(),
+              activeParamIndices));
+
+           // Check and diagnose non-differentiable original function type.
```

Comment:
This is not diagnosing. Please update names to match its new semantics when reorganizing code.